openedx/adg/admin_dashboard/performance/views.py

```diff
@@ -1,3 +1,151 @@
+
+        Returns:
+            bool: True if user has all permissions, False otherwise
+        """
+        user = self.request.user
+        self.course = self._get_course()
+
+        if not (user.is_authenticated and user.is_staff):
+            return False
+
+        if user.is_superuser or is_adg_admin(user):
+            return True
+
+        if not user.groups.filter(id__in=BusinessLine.objects.values_list('group__id')).exists():
+            return False
+
+        return MultilingualCourseGroup.objects.filter(
+            multilingual_courses__course__id=self.course.id
+        ).filter(
+            Q(is_common_business_line_prerequisite=True) | Q(business_line_prerequisite__group__in=user.groups.all())
+        ).exists()
+
+    def get_context_data(self, *, object_list=None, **kwargs):
+        """
+        Overridden to add course grades for the users.
+
+        Returns:
+            dict: Dict containing context data.
+        """
+        object_list = self.get_queryset().only('id', 'email', 'profile__name')
+        context_data = super().get_context_data(object_list=object_list)
+
+        user_ids = [user.id for user in context_data['object_list']]
+        course_grades = get_course_grades_for_users(user_ids, self.course.id)
+        for user in context_data['object_list']:
+            user.course_grade = course_grades[user.id]
+
+        return context_data
+
+    def get(self, request, *args, **kwargs):
+        """
+        Overridden to add extra context data.
+        """
+        self.extra_context = {
+            'course_name': self.course.display_name,
+        }
+        return super().get(request, *args, **kwargs)
+
+    def post(self, request, *args, **kwargs):
+        """
+        Attaches csv report for download with response.
+
+        Returns:
+            HttpResponse: Response with grades report.
+        """
+        users = self.get_queryset().only('id', 'username', 'email', 'profile__phone_number', 'profile__name',)
+        course_grades = get_course_grades_for_users(users.values_list('id'), self.course.id)
+
+        report_data = get_course_grades_report_data(users, course_grades)
+        buf = StringIO()
+        writer = csv.writer(buf)
+        writer.writerows(report_data)
+
+        response = HttpResponse(buf.getvalue(), content_type='text/csv')
+        response['Content-Disposition'] = f'attachment; filename={self.course.display_name}_{date.today()}_grades.csv'
+        return response
+
+    def get_queryset(self):
+        """
+        Overridden to custom filters.
+
+        Returns
+            queryset: Queryset containing enrolled users of the course.
+        """
+        course_enrollments_prefetch_object = Prefetch(
+            'courseenrollment_set',
+            queryset=CourseEnrollment.objects.filter(
+                course_id=self.course.id).only(
+                'id', 'created', 'user_id'
+            ),
+            to_attr='course_enrollment'
+        )
```

Comment:
Have we profiled this prefetch? Does adding it save us any queries? 
I ask this because prefetch will immediately load all this data into memory (no lazy loading), and CourseEnrollment data might be a huge amount. Just highlighting a small concern, not really a suggestion.